### PR TITLE
Fix stacking of pos/neg logits for RM compute_metrics

### PR DESCRIPTION
**Pull request by andreaskoepf**
Node positive & negative examples are stacked along last dimension so that compute_metrics receives predictions of size (val_examles, 2). The old code did not work with deepspeed if the number of validation examples was not divisible by the total batch size without remainder.

### CODE REVIEW

**Review by sanagno — APPROVED** [review]

**Review by yk — APPROVED** [review]
Thank you
